filebeat/harvester/registry.go

```diff
@@ -39,9 +45,7 @@ func (r *Registry) Stop() {
 	close(r.done)
 
 	for _, hv := range r.harvesters {
-		r.wg.Add(1)
 		go func(h Harvester) {
-			r.wg.Done()
 			h.Stop()
 		}(hv)
 	}
```

Comment:
The difference is when it's blocking and when the mutex is released. My proposal releases the mutex only after all harvesters are finished -> start can only be called after Stop did return.